healpy/fitsfunc.py

```diff
@@ -294,7 +314,9 @@ def read_map(
       Force the conversion to some type. Passing a list allows different
       types for each field. In that case, the length of the list must
       correspond to the length of the field parameter. Default: np.float64
-      if None, keep the dtype of the input FITS file
```

Comment:
the issue here is that we are loosing a functionality, which is maintaining the dtype of the input file, which I think is very useful.
If the default is to use `np.float64`, I think it is better to show it into the `read_map` signature directly as it is currently.

I would leave `read_map` as it is and only modify `read_cl` and `write_cl`

Reply:
I think I understand ... but then it becomes really tricky to decide when the FutureWarning should be printed.
In the future all defaults are supposed to change to `None`, of course.

Reply:
OK, I think I found a solution. But it's incredibly ugly...

Reply:
yeah, too ugly, I would leave it as it was.
As long as things are documented and do not change defaults from one version to another, I don't think inconsistent is a big issue.

Reply:
Sorry, I have to disagree with this statement. An inconsistency in the interface is always a burden on the user: they have to keep in mind that they need to use some function with special caution, because there is some inconsistency. But on the other hand, default arguments are supposed to make things _easier_ for the user. This is clearly not happening here.

I'm perfectly fine with simply removing all the `dtype` defaults. That would in fact be my favorite option.

Concerning the ugliness: it would of course go away once the switch to the new interface is complete.

Reply:
I think we can keep the current behavior *and* print the `FutureWarning`, later on, we will make the default to have `dtype=None` and keep whatever `dtype` is in the file.

Reply:
OK. Does this look better now?

Reply:
I am worried people won't see the warning in the docstring.
If think it is useful to check if the input `dtype` was `np.float64` and in that case print a `FutureWarning` that explains that 

> If you are not specifying the input dtype and using the default `np.float64` dtype of read_map(), please consider that it will change in a future version to None as to keep the same dtype of the input file: please explicitly set the dtype if it is important to you

Unfortunately also people that specify `np.float64` will get this warning, but no easy way around it, and better be more verbose then not.

Reply:
Well, that was exactly the reason for the "ugly" default of 42, which would have allowed us to discern between the user explicitly specifying `np.float64` or nothing at all.
Would things be better if I replaced `42` by some opaque dummy value like `healpy.NOT_SET`?

Reply:
no I'd rather print an extra warning that people can easily disable than implement some clever workaround